backend/akr/src/main/java/fi/oph/akr/audit/dto/ClerkTranslatorAuditDTO.java

```diff
@@ -0,0 +1,53 @@
+package fi.oph.akr.audit.dto;
+
+import fi.oph.akr.api.dto.translator.TranslatorAddressDTO;
+import fi.oph.akr.model.Translator;
+import fi.oph.akr.onr.model.PersonalData;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import lombok.Builder;
+import lombok.NonNull;
+
+@Builder
+public record ClerkTranslatorAuditDTO(
+  @NonNull Long id,
+  @NonNull Integer version,
+  @NonNull Boolean isIndividualised,
+  @NonNull Boolean hasIndividualisedAddress,
+  @NonNull String firstName,
+  @NonNull String lastName,
+  @NonNull String nickName,
+  String identityNumber,
+  String email,
+  String phoneNumber,
+  Map<String, TranslatorAddressDTO> address,
+  String extraInformation,
+  @NonNull Boolean isAssuranceGiven,
+  String selectedSource,
+  String selectedType
+)
+  implements AuditEntityDTO {
+  public ClerkTranslatorAuditDTO(Translator translator, PersonalData personalData) {
+    this(
+      translator.getId(),
+      translator.getVersion(),
+      personalData.getIndividualised(),
+      personalData.getHasIndividualisedAddress(),
+      personalData.getFirstName(),
+      personalData.getLastName(),
+      personalData.getNickName(),
+      personalData.getIdentityNumber(),
+      personalData.getEmail(),
+      personalData.getPhoneNumber(),
+      personalData
+        .getAddress()
+        .stream()
+        .collect(Collectors.toMap(address -> address.source() + ":" + address.type(), Function.identity())),
```

Comment:
Tämä on nyt tehty olettaen että source + type on uniikki avain. Tämä heittää poikkeuksen jos näin ei joskus ole - mikä on ehkä hyvä että jäisi haaviin jos oletus on väärä.